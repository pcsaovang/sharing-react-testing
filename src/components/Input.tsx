import classnames from "classnames";

type Props = React.InputHTMLAttributes<HTMLInputElement> & {};

function Input(props: Props): React.ReactElement {
  const buttonClass = classnames(
    "py-2 px-3",
    "border",
    "border-gray-300",
    "focus:border-red-300",
    "focus:outline-none",
    "focus:ring",
    "focus:ring-red-200",
    "focus:ring-opacity-50",
    "rounded-md",
    "shadow-sm",
    "disabled:bg-gray-100",
    "mt-1",
    "block w-full"
  );

  return <input className={buttonClass} {...props} />;
}

export default Input;
