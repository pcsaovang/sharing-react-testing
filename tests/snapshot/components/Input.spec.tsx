import { render } from "@testing-library/react";
import Input from "../../../src/components/Input";

describe("Input component", () => {
  it("Render Input with default props", () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });
});
