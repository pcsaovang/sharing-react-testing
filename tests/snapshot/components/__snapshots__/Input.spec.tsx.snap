// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Render Input with default props 1`] = `
<div>
  <input
    class="py-2 px-3 border border-gray-300 focus:border-red-300 focus:outline-none focus:ring focus:ring-red-200 focus:ring-opacity-50 rounded-md shadow-sm disabled:bg-gray-100 mt-1 block w-full"
  />
</div>
`;
